File: src/main/java/com/learn/todolist/api/controller/TodoController.java
package com.learn.todolist.api.controller;

import com.learn.todolist.api.dto.TodoCreateReq;
import com.learn.todolist.api.dto.TodoModifyReq;
import com.learn.todolist.application.CommonResponse;
import com.learn.todolist.application.TodoApplication;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RequiredArgsConstructor
@RestController
@RequestMapping("/api/v1/todos")
public class TodoController {
    private final TodoApplication todoApplication;

    @GetMapping
    public ResponseEntity<CommonResponse> getTodoList() {
        CommonResponse response = todoApplication.processGetTodoList();
        return ResponseEntity.ok(response);
    }

    @PostMapping("/todo")
    public ResponseEntity<CommonResponse> createTodo(@RequestBody TodoCreateReq request) {
        CommonResponse response = todoApplication.processCreateTodo(request);
        return ResponseEntity.ok(response);
    }

    @PatchMapping("/todo/{id}")
    public ResponseEntity<CommonResponse> updateTodo(@PathVariable Long id, @RequestBody TodoModifyReq request) {
        CommonResponse response = todoApplication.processUpdateTodo(id, request);
        return ResponseEntity.ok(response);
    }

    @DeleteMapping("/todo/{id}")
    public ResponseEntity<CommonResponse> deleteTodo(@PathVariable Long id) {
        CommonResponse response = todoApplication.processDeleteTodo(id);
        return ResponseEntity.ok(response);
    }

    @GetMapping("/todo/{id}")
    public ResponseEntity<CommonResponse> getTodoDetail(@PathVariable Long id) {
        CommonResponse response = todoApplication.processGetTodoDetail(id);
        return ResponseEntity.ok(response);
    }

}

File: src/main/kotlin/com/todolist/aplication/todo/controller/TodoController.kt
package com.todolist.aplication.todo.controller

import com.todolist.aplication.todo.dto.request.TodoCreateRequest
import com.todolist.aplication.todo.dto.request.TodoUpdateRequest
import com.todolist.aplication.todo.usecase.TodoApplication
import com.todolist.domain.common.CommonResponse
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1")
class TodoController(
    private val todoApplication: TodoApplication
)   {
    @PostMapping("/todos/todo")
    fun createTodo(@RequestBody request: TodoCreateRequest): ResponseEntity<CommonResponse> {
        val response = todoApplication.processCreateTodo(request)
        return ResponseEntity(response, HttpStatus.OK)
    }

    @PatchMapping("/todos/{id}")
    fun updateTodo(@RequestBody request: TodoUpdateRequest): ResponseEntity<CommonResponse> {
        val response = todoApplication.processUpdateTodo(request)
        return ResponseEntity(response, HttpStatus.OK)
    }

    @DeleteMapping("/todos/{id}")
    fun deleteTodo(@PathVariable id: String): ResponseEntity<CommonResponse> {
        val response = todoApplication.processDeleteTodo(id)
        return ResponseEntity(response, HttpStatus.OK)
    }

    @GetMapping("/todos")
    fun getAllTodo(): ResponseEntity<CommonResponse> {
        val response = todoApplication.processGetAllTodo()
        return ResponseEntity(response, HttpStatus.OK)
    }

    @GetMapping("/todos/{id}")
    fun getTodoDetail(@PathVariable id: String): ResponseEntity<CommonResponse> {
        val response = todoApplication.processGetTodoDetail(id)
        return ResponseEntity(response, HttpStatus.OK)
    }

    @GetMapping("/todos/recent-todo")
    fun getRecentTodo(): ResponseEntity<CommonResponse> {
        val response = todoApplication.processGetRecentTodo()
        return ResponseEntity(response, HttpStatus.OK)
    }


}